src/spellsuggest.c

```diff
@@ -1619,7 +1619,7 @@ suggest_trie_walk(
 		    // char, e.g., "thes," -> "these".
 		    p = fword + sp->ts_fidx;
 		    MB_PTR_BACK(fword, p);
-		    if (!spell_iswordp(p, curwin) && *preword != NUL)
+		    if (!spell_iswordp(p, curwin) && *preword != NULL)
```

Comment:
`NUL` is correct. `NUL` is defined as `'\000'` in ascii.h.
`preword` is an array of `char_u`, not an array of pointer.
So, using `NULL` here is wrong.